tests/Shared/SolutionTests.proj

```diff
@@ -0,0 +1,30 @@
+<Project Sdk="Microsoft.Build.Traversal">
+  <!--
+    This file is used to build all the eligible test projects in the repo. 
+    It is used to execute the tests in the outer loop and in the CI pipeline.
+
+    Generally, this project should invoked in the following way:
+
+      ./build.cmd -projects ./tests/Shared/SolutionTests.proj -restore -build -test /p:IsGitHubActions=true /p:RunQuarantinedTests=[true|false]
```

Comment:
Can you help me understand a bit better why this is needed? I think that running `build.cmd -restore -build -test ...` already works and will test all the projects in the repo as we have this: https://github.com/dotnet/aspire/blob/fca6ddb80eaca68d0c694c187223c7006b8d6119/eng/Build.props#L7. If what you want is to make it so we only build tests and not the product, feels like we could reuse that Build.props and just add conditions that can be set via build.cmd command for this without having to introduce a new file. As an example, we already have `build.cmd -build -testnobuild` which builds all of the product but not the tests.

Reply:
The reason I created a dedicated proj file was to execute only the required test projects, following the established filtering pattern in 
https://github.com/dotnet/aspire/blob/b3cc78f869a2a709454c78bac7d2ebe228e6f949/tests/Shared/GetTestProjects.proj#L17-L27

I treat this is an intermediary step to address the immediate task. The current implementations are currently a) fluid and b) complicated, and it doesn't feel appropriate to splice in these.

I'm not 100% clear on the target state, but I think if each test project declared what test infra it's expected to be executed (e.g., by a virtue of having an MSBuild property), it could allow us to simplify the current infra.